sys/net/gnrc/network_layer/ipv6/nib/Kconfig

```
@@ -0,0 +1,108 @@
+
+config GNRC_IPV6_NIB_6LR
+    bool "6Lo router features"
+    default y if MODULE_GNRC_IPV6_NIB_6LR || GNRC_IPV6_NIB_CONF_6LBR
+
+config GNRC_IPV6_NIB_6LN
+    bool "6Lo node features"
+    default y if MODULE_GNRC_IPV6_NIB_6LN || GNRC_IPV6_NIB_6LR
+
+config GNRC_IPV6_NIB_ROUTER
+    bool "Router features"
+    default y if MODULE_GNRC_IPV6_NIB_ROUTER || GNRC_IPV6_NIB_6LR
+
+config GNRC_IPV6_NIB_SLAAC
+    bool "Stateless address auto-configuration"
+    default y if MODULE_GNRC_IPV6_NIB_6LBR
+    default n if MODULE_GNRC_IPV6_NIB_6LR || MODULE_GNRC_IPV6_NIB_6LN
+    default y
+
+config GNRC_IPV6_NIB_QUEUE_PKT
+    bool "Queue packets"
+    default n if MODULE_GNRC_IPV6_NIB_6LN || GNRC_IPV6_NIB_6LN
+    default y
+
+config GNRC_IPV6_NIB_ARSM
+    bool "NDP address resolution state-machine"
+    default n if MODULE_GNRC_IPV6_NIB_6LN && !GNRC_IPV6_NIB_6LR
+    default y
+
+config GNRC_IPV6_NIB_DNS
+    bool "Support for DNS configuration options"
+    default y if MODULE_GNRC_IPV6_NIB_DNS
+
+config GNRC_IPV6_NIB_ADV_ROUTER
+    bool "Router adversiting"
+    default y if GNRC_IPV6_NIB_ROUTER && (!GNRC_IPV6_NIB_6LR || GNRC_IPV6_NIB_CONF_6LBR)
+
+config GNRC_IPV6_NIB_REDIRECT
+    bool "Handle redirect messages"
+
+config GNRC_IPV6_NIB_DC
+    bool "Destination cache"
+    default y if GNRC_IPV6_NIB_REDIRECT
+
+config GNRC_IPV6_NIB_MULTIHOP_DAD
+    bool "Multihop duplicate address detection"
+    help
+        @see [RFC 6775, section 8.2](https://tools.ietf.org/html/rfc6775#section-8.2).
+
+config GNRC_IPV6_NIB_MULTIHOP_P6C
+    bool "Multihop prefix and 6LoWPAN context distribution"
+    default y if GNRC_IPV6_NIB_6LR
+
+config GNRC_IPV6_NIB_NO_RTR_SOL
+    bool "Disable router solicitations"
+    help
+        @warning Only set this if you know what you're doing.
+
+config GNRC_IPV6_NIB_NUMOF
+    int "Number of entries in NIB"
+    default 16 if MODULE_GNRC_IPV6_NIB_6LBR
+    default 1 if MODULE_GNRC_IPV6_NIB_6LN && !GNRC_IPV6_NIB_6LR
+    default 4
+
+config GNRC_IPV6_NIB_REACH_TIME_RESET
+    int "Reset time for the reachability time (milliseconds)"
+    default 7200000
+    help
+        @see [RFC 4861, section 6.3.4](https://tools.ietf.org/html/rfc4861#section-6.3.4).
+
+config GNRC_IPV6_NIB_L2ADDR_MAX_LEN
+    int "Maximum link-layer address length (aligned)"
+    default 8
+
+config GNRC_IPV6_NIB_DEFAULT_ROUTER_NUMOF
+    int "Number of default routers in the default router list"
+    default 1
+    help
+        @attention This number has direct influence on the maximum number of
+        neighbors and duplicate address detection table entries.
+
+config GNRC_IPV6_NIB_OFFL_NUMOF
+    int "Number of off-link entries in NIB"
+    default 8
+    help
+        @attention This number is equal to the maximum number of forwarding
+        table and prefix list entries in NIB.
+
+config GNRC_IPV6_NIB_ABR_NUMOF
+    int "Number of authoritative border router entries in NIB"
+    default 1
+    depends on GNRC_IPV6_NIB_MULTIHOP_P6C
```

I know this was not documented previously, but now that this is more prominent, I think this is worth pointing out:

```suggestion
    depends on GNRC_IPV6_NIB_MULTIHOP_P6C
    help
        @warning Behavior for >2 currently not specified or tested.
```